lalrpop/src/lr1/codegen/parse_table.rs

```diff
@@ -776,16 +776,15 @@ impl<'ascent, 'grammar, W: Write> CodeGenerator<'ascent, 'grammar, W, TableDrive
     fn emit_downcast_fn(&mut self, variant_name: &str, variant_ty: TypeRepr) -> io::Result<()> {
         let spanned_symbol_type = self.spanned_symbol_type();
 
-        rust!(self.out, "fn {}pop_{}<", self.prefix, variant_name);
-        for type_parameter in &self.custom.symbol_type_params {
-            rust!(self.out, "  {},", type_parameter);
-        }
-        rust!(self.out, ">(");
-        rust!(self.out,
-              "{}symbols: &mut ::std::vec::Vec<{}>",
-              self.prefix,
-              spanned_symbol_type);
-        rust!(self.out, ") -> {} {{", self.types.spanned_type(variant_ty));
+        try!(self.out.write_fn_header(&self.grammar,
```

Comment:
Also, yes, to fix your bug, we will have to add some code to the call-site to thread the parameters through explicitly, like this code here: https://github.com/nikomatsakis/lalrpop/blob/master/lalrpop/src/lr1/codegen/parse_table.rs#L689-L697

So basically we want to change [this call](https://github.com/nikomatsakis/lalrpop/blob/master/lalrpop/src/lr1/codegen/parse_table.rs#L632) to include the `::<{}>` with the value [from here](https://github.com/nikomatsakis/lalrpop/blob/master/lalrpop/src/lr1/codegen/parse_table.rs#L695)